src/partition/repart.c

```diff
@@ -7392,6 +7435,62 @@ static int context_crypttab(Context *context) {
         return 0;
 }
 
+/* update block sizes for verity siblings, calculate hash partition size if requested */
+static int context_update_verity_size(Context *context) {
+        int r;
+
+        assert(context);
+
+        LIST_FOREACH(partitions, p, context->partitions) {
+                Partition *dp;
+
+                if (p->verity != VERITY_HASH)
+                        continue;
+
+                if (p->dropped)
+                        continue;
+
+                if (PARTITION_EXISTS(p)) /* Never format existing partitions */
+                        continue;
+
+                assert_se(dp = p->siblings[VERITY_DATA]);
+
+                if (p->verity_data_block_size == UINT64_MAX)
+                        p->verity_data_block_size = context->fs_sector_size;
+
+                if (p->verity_hash_block_size == UINT64_MAX)
+                        p->verity_hash_block_size = context->fs_sector_size;
+
+                uint64_t sz;
+                if (dp->size_max != UINT64_MAX) {
+                        r = calculate_verity_hash_size(
+                                        dp->size_max,
+                                        p->verity_hash_block_size,
+                                        p->verity_data_block_size,
+                                        &sz);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to caculate size of dm-verity hash partition");
+
+                        if (sz > p->size_max) {
+                                log_warning("SizeMaxBytes=%" PRIu64 " of dm-verity hash partition too small for "
+                                            "data partition with SizeMaxBytes=%" PRIu64 ". Hash partition would "
+                                            "require %" PRIu64 " bytes for a data partition of specified max size. "
+                                            "Consider increasing SizeMaxBytes= of hash partition, or decreasing "
+                                            "SizeMaxBytes= of data partition.",
+                                            p->size_max, dp->size_max, sz);
+                        } else {
```

Comment:
nitpick: no {} around singe line if blocks, i.e. drop the {} around the first if branch here